### ISSUE TITLE
Feature: Record only when actually Recording

### ISSUE THREAD

**Issue by armand1m** [perry-js/perry#12]
For now Perry is recording every single click of the application when it is instantiated.

This behaviour is due to the hooks set in these lines:
https://github.com/perry-js/perry/blob/master/src/packages/perry/index.ts#L57-L59

Now, these hooks should move to the `Widget` component. It can be assigned to it through the props object in the same file, at the `render()` method.

Instead of just:
```js
    const props: WidgetProps = {
      onSubmit: (reportInfo: PerryReportInfo) => {
        const report = aggregateReport(reportInfo);

        options.plugins.map(plugin => plugin(report));

        return report;
      }
    };
```

It will be:

```js
    const props: WidgetProps = {
      onStartRecording: () => {
        // subscribe
        applyConsoleProxy(options);
        listenWindowErrors(options);
        listenDocumentClicks(options);
      },
      onStopRecording: () => {
        // unsubscribe
        removeConsoleProxy(options);
        removeWindowErrorsListener(options);
        removeDocumentClicksListener(options);
      },
      onSubmit: (reportInfo: PerryReportInfo) => {
        const report = aggregateReport(reportInfo);

        options.plugins.map(plugin => plugin(report));

        return report;
      }
    };
```

Basically these hooks need to be modified to allow them to be unsubscribed as well. The following methods will need to be created within the PR:

```js
removeConsoleProxy(options);
removeWindowErrorsListener(options);
removeDocumentClicksListener(options);
```

Also, the `<Widget />` component will have to adapt to these new props, and actually run the subscription handlers when it changes its states in the `nextStep` instance method.

The `WidgetProps`  interface will also have to change.

https://github.com/perry-js/perry/blob/master/src/components/Widget/index.tsx

**Comment by armand1m** [comment]
@vitormargis I'll implement the `removeConsoleProxy(options);` method now and submit a pull request =)
